studio/LonaStudio/Workspace/FileNavigator.swift

```diff
@@ -185,6 +185,36 @@ class FileNavigator: NSBox {
                 menu.addItem(NSMenuItem.separator())
             }
 
+            if NSURL(fileURLWithPath: path).pathExtension == "component"{
+                menu.addItem(NSMenuItem(title: "Duplicate As...", onClick: {
+                    var saveURL: String
+                    let dialog = NSSavePanel()
+
+                    dialog.title                   = "Save .component file"
+                    dialog.showsResizeIndicator    = true
+                    dialog.showsHiddenFiles        = false
+                    dialog.canCreateDirectories    = true
+                    dialog.allowedFileTypes        = ["component"]
+
+                    // User canceled the save. Don't swap out the document.
+                    if dialog.runModal() != NSApplication.ModalResponse.OK {
+                        return
+                    }
+
+                    guard let url = dialog.url else { return }
+                    saveURL = url.path
```

Comment:
I would either: 
- remove the `saveURL` variable and call `url.path` directly when we need it (simpler, 1 fewer var) 
- move the variable declaration down to the same line (`let saveURL = url.path`) and rename `saveURL` to `savePath`, since it's not actually a `URL` anymore